src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -1291,6 +1417,71 @@ def construct_payload(self):
+            existing_registries = [r for r in existing_registries if ACR_IMAGE_SUFFIX in r["server"]]
+            if existing_registries is None or len(existing_registries) == 0:
+                raise ValidationError(
+                    "Error: The containerapp '{}' does not have an ACR registry associated with it. Please specify a registry using the --registry-server argument while creating the ContainerApp".format(
+                        self.get_argument_name()))
+            registry_server = existing_registries[0]["server"]
+            parsed = urlparse(registry_server)
+            registry_name = (parsed.netloc if parsed.scheme else parsed.path).split('.')[0]
+            registry_user, registry_pass, _ = _get_acr_cred(self.cmd.cli_ctx, registry_name)
+            self._update_container_app_source(cmd=self.cmd, registry_server=registry_server, registry_user=registry_user, registry_pass=registry_pass)
+
+    def _update_container_app_source(self, cmd, registry_server, registry_user, registry_pass):
+        from ._up_utils import (ContainerApp, ResourceGroup, ContainerAppEnvironment, _reformat_image, _has_dockerfile, _get_dockerfile_content, _get_ingress_and_target_port, _get_registry_details)
+
+        ingress = self.get_argument_ingress()
+        target_port = self.get_argument_target_port()
+        dockerfile = "Dockerfile"
+
+        # Parse resource group name and managed env name
+        env_id = self.containerapp_def["properties"]['environmentId']
+        parsed_managed_env = parse_resource_id(env_id)
+        env_name = parsed_managed_env['name']
+        env_rg = parsed_managed_env['resource_group']
+
+        # Set image to None if it was previously set to the default image (case where image was not provided by the user) else reformat it
+        image = None if self.get_argument_image() is None else _reformat_image(source=self.get_argument_source(), image=self.get_argument_image(), repo=None)
+        location = self.containerapp_def["location"]
+
+        has_dockerfile = _has_dockerfile(self.get_argument_source(), dockerfile)
+        if has_dockerfile:
+            dockerfile_content = _get_dockerfile_content(repo=None, branch=None, token=None, source=self.get_argument_source(), context_path=None, dockerfile=dockerfile)
+            ingress, target_port = _get_ingress_and_target_port(self.get_argument_ingress(), self.get_argument_target_port(), dockerfile_content)
+
+        # Construct ContainerApp
+        resource_group = ResourceGroup(cmd, env_rg, location=location)
+        env = ContainerAppEnvironment(cmd, env_name, resource_group, location=location)
+        app = ContainerApp(cmd=cmd, name=self.get_argument_name(), resource_group=resource_group, image=image, env=env, target_port=target_port, workload_profile_name=self.get_argument_workload_profile_name(), ingress=ingress, registry_server=registry_server, registry_user=registry_user, registry_pass=registry_pass)
```

Comment:
Here uses env's rg value to create `resource_group`, but we accept containerapp and environment in different resource group.
You should use `self.get_argument_resource_group_name()` as `rg `value for ContainerApp.

Reply:
Makes sense! Updated this.